tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -5055,7 +5053,10 @@ class HLSLExternalSource : public ExternalSemaSource {
 
   bool CheckRangedTemplateArgument(SourceLocation diagLoc,
                                    llvm::APSInt &sintValue) {
-    if (!sintValue.isStrictlyPositive() || sintValue.getLimitedValue() > 4) {
+    const auto *SM =
+        hlsl::ShaderModel::GetByName(m_sema->getLangOpts().HLSLProfile.c_str());
+    if (!sintValue.isStrictlyPositive() ||
+        (sintValue.getLimitedValue() > 4 && !SM->IsSM69Plus())) {
```

Comment:
I'll leave my objection here:
This changes what can be parsed globally, not just based on what's used by an entry point being compiled. In my opinion, switching global HLSL language availability based on shader model is ugly and unexpected.

Reply:
One point on this code though:
Shouldn't we have something like a `MaximumVectorLength` value, which is checked here, and can be used with an updated diagnostic message?